### ISSUE TITLE
cmd/compile: similar returns not optimized

### ISSUE THREAD

**Issue by marigonzes** [golang/go#32554]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version devel +323212b Sun Jun 9 16:23:11 2019 +0000 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes.

### What did you do?
I compiled the following function:
```go
func f1(x int, b bool) int {
	if b {
		return x
	}
	return x
}
```

### What did you expect to see?
I expected it to be compiled to a move and a return, given that the return value is always the same.

### What did you see instead?
Instead, it was compiled to:
```asm
        movblzx "".b+16(SP), AX
        testb   AL, AL
        jeq     f1_pc20
        movq    "".x+8(SP), AX
        movq    AX, "".~r2+24(SP)
        ret
f1_pc20:
        movq    "".x+8(SP), AX
        movq    AX, "".~r2+24(SP)
        ret
```
I think the compiler should recognize both cases to be the same.

Note: if this function is called from another one, the optimization is applied. For example:
```go
func f2(x int, b bool) int {
    return f1(x, b)
}
```
is compiled to
```asm
        movq    "".x+8(SP), AX
        movq    AX, "".~r2+24(SP)
        ret
```

**Comment by randall77** [comment]
Yep, this is a TODO. I even have a CL for it: https://go-review.googlesource.com/c/go/+/33634
I never got around to actually finishing it up.

**Comment by MichaelTJones** [comment]
Is the underlying issue different than the split/common approach in your CL? That seems a late stage remedy to what should already have been rewritten away by a "Moot Code Elimination" pass.
